### PR TITLE
doc: update the doc of the --log_rotate_max_age

### PR DESCRIPTION
**Pull request by doujiang24**

### CODE REVIEW

File: pkg/log/options.go [hunk 1]
```diff
@@ -84,12 +84,12 @@ type Options struct {
 	ErrorOutputPaths []string
 
 	// RotateOutputPath is the path to a rotating log file. This file should
-	// be automatically rotated over time, based on the rotation parameters such
-	// as RotationMaxSize and RotationMaxAge. The default is to not rotate.
+	// be automatically rotated over time, based on the rotation parameter:
+	// RotationMaxSize.
```
Comment (wulianglongrd) [inline pkg/log/options.go:88]: why delete `RotationMaxAge`?

Comment (doujiang24) [reply pkg/log/options.go:88]: RotationMaxAge does not affect file rotation, just affect file backups instead.

Comment (doujiang24) [reply pkg/log/options.go:88]: Any more questions? cc @hzxuzhonghu 

The comments is wrong, according to [lumberjack](https://github.com/natefinch/lumberjack/blob/v2.0/lumberjack.go#L89-L94):
```
// MaxAge is the maximum number of days to retain old log files based on the
// timestamp encoded in their filename
```

**Comment by hzxuzhonghu** [comment]
/ok-to-test

sorry i missed it

File: pkg/log/options.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/log/options.go:88]: Donot we call storeing logs with timestamp as rotation? I think this is also a kind of rotation

Comment (doujiang24) [reply pkg/log/options.go]: okay, seems make sense too, just reverted it.

**Comment by doujiang24** [comment]
CI failed due to this error:
```
ERROR: Missing release notes and missing "release-notes-none" label.
```
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/50662/release-notes_istio/1795457642357854208#1:build-log.txt%3A31

@hzxuzhonghu Could you please add the `release-notes-none` label? This is just a doc fix, no release note required. Thanks~

**Review by hzxuzhonghu — APPROVED** [review]
Thanks

**Review by howardjohn — APPROVED** [review]

**Pull request merged**